app/utils/skills-list.js

```diff
@@ -0,0 +1,49 @@
+import Ember from 'ember';
+
+const {
+  get
+} = Ember;
+
+let skillsList = {
+  find(skills, skill, relationship) {
+    if (skills) {
+      let userSkill = skills.find((item) => {
+        let model = get(item, 'constructor.modelName');
+        return _matchForModel(model, item, skill, relationship);
+      });
+      return userSkill;
+    }
+  },
+
+  contains(skills, skill) {
+    if (skills) {
+      let matchedSkill = skills.find((item) => {
+        let itemSkillId = item.belongsTo('skill').id();
+        let skillId = get(skill, 'id');
+        return (itemSkillId === skillId);
+      });
+      return !isEmpty(matchedSkill);
+    } else {
+      false;
+    }
+  }
+}
+
+function _matchForModel(model, item, skill, relationship) {
+  switch(model) {
+    case 'user-skill':
+      return _matchWithRelationship(item, skill, relationship, 'user');
```

Comment:
This `switch` was best way I could think of. Would appreciate other suggestions.